Data/Sys/Shaders/default_pre_post_process.glsl

```diff
@@ -4,92 +4,152 @@
+    0.000000000000000, 0.000000000000000, 0.988195521789351);
 
-float3 LinearTosRGBGamma(float3 color)
+/***** Linear <--> sRGB *****/
+
+float4 LinearTosRGB(float4 c)
+{
+    if (!OptionEnabled(correct_gamma)) {
+        return pow(c, float4(1.0 / game_gamma));
+    }
+
+    else if (OptionEnabled(sdr_display_gamma_sRGB)) {
+        const float a = 0.055;
+
+        c.r = (c.r < 0.0031308) ? (12.92 * c.r) : (pow(c.r, 1.0 / 2.4) * (1.0 + a) - a);
+        c.g = (c.g < 0.0031308) ? (12.92 * c.g) : (pow(c.g, 1.0 / 2.4) * (1.0 + a) - a);
+        c.b = (c.b < 0.0031308) ? (12.92 * c.b) : (pow(c.b, 1.0 / 2.4) * (1.0 + a) - a);
+
+        return c;
+    }
+
+    else {
+        return pow(c, float4(1.0 / sdr_display_custom_gamma));
+    }
+}
+
+float4 sRGBToLinear(float4 c)
+{
+    // If game is using sRGB
+    if (false) {
+        const float a = 0.055;
+
+        c.r = (c.r <= 0.04045) ? (c.r / 12.92) : pow((c.r + a) / (1.0 + a), 2.4);
+        c.g = (c.g <= 0.04045) ? (c.g / 12.92) : pow((c.g + a) / (1.0 + a), 2.4);
+        c.b = (c.b <= 0.04045) ? (c.b / 12.92) : pow((c.b + a) / (1.0 + a), 2.4);
+
+        return c;
+    }
+
+    else {
+        return pow(c, float4(game_gamma));
+    }
+}
+
+/***** Linear <--> Oklab *****/
+
+const mat4 RGBtoLMS = mat4(
+    0.4122214708, 0.2119034982, 0.0883024619, 0.0000000000,
+    0.5363325363, 0.6806995451, 0.2817188376, 0.0000000000,
+    0.0514459929, 0.1073969566, 0.6299787005, 0.0000000000,
+    0.0000000000, 0.0000000000, 0.0000000000, 1.0000000000);
+
+const mat4 LMStoOklab = mat4(
+    0.2104542553, 1.9779984951, 0.0259040371, 0.0000000000,
+    0.7936177850, -2.4285922050, 0.7827717662, 0.0000000000,
+    -0.0040720468, 0.4505937099, -0.8086757660, 0.0000000000,
+    0.0000000000, 0.0000000000, 0.0000000000, 1.0000000000);
+
+float4 LinearRGBToOklab(float4 c)
 {
-	const float a = 0.055;
-
-	for (int i = 0; i < 3; ++i)
-	{
-		float x = color[i];
-		if (x <= 0.0031308)
-			x = x * 12.92;
-		else
-			x = (1.0 + a) * pow(x, 1.0 / 2.4) - a;
-		color[i] = x;
-	}
-
-	return color;
+    return LMStoOklab * pow(RGBtoLMS * c, float4(1.0 / 3.0));
 }
 
-/***** COLOR SAMPLING *****/
+const mat4 OklabtoLMS = mat4(
+    1.0000000000, 1.0000000000, 1.0000000000, 0.0000000000,
+    0.3963377774, -0.1055613458, -0.0894841775, 0.0000000000,
+    0.2158037573, -0.0638541728, -1.2914855480, 0.0000000000,
+    0.0000000000, 0.0000000000, 0.0000000000, 1.0000000000);
+
+const mat4 LMStoRGB = mat4(
+    4.0767416621, -1.2684380046, -0.0041960863, 0.0000000000,
+    -3.3077115913, 2.6097574011, -0.7034186147, 0.0000000000,
+    0.2309699292, -0.3413193965, 1.7076147010, 0.0000000000,
+    0.0000000000, 0.0000000000, 0.0000000000, 1.0000000000);
+
+float4 OklabToLinearRGB(float4 c)
+{
+    return max(LMStoRGB * pow(OklabtoLMS * c, float4(3.0)), 0.0);
+}
+
+/***** Quick Sampling *****/
 
 // Non filtered gamma corrected sample (nearest neighbor)
-float4 QuickSample(float3 uvw, float gamma)
```

Comment:
I'm not a big fan of removing the gamma parameter from samples and instead reading it from the global parameter.
Performance wise, nothing changes as shader functions are all inlined when compiled, but the functions are less generic and re-usable now, which I don't like. Unnecessary change IMO.

Reply:
I would agree, I'd like to keep these reusable if possible